libs/domain/picking/src/components/picking-header/picking-header.component.ts

```diff
@@ -55,12 +73,20 @@ export class PickingHeaderComponent extends PickingListMixin(LitElement) {
           triggerType: 'icon',
           contentBehavior: 'modal',
         }}
-      ></oryx-site-navigation-item>`;
+      ></oryx-site-navigation-item>
+      <oryx-discard-picking
+        ${ref(this.discardModal)}
+        @oryx.back=${this.back}
+      ></oryx-discard-picking>`;
+  }
+
+  protected openDiscardModal(): void {
+    const modal = this.discardModal.value;
+    modal && (modal.open = true);
   }
 
   protected back(): void {
-    //TODO - display discard modal
-    this.routerService.back();
+    this.routerService.back(this.pickingList?.cartNote ? 2 : 1);
```

Comment:
looks not very stable. I believe we should find exact step, where we would like to navigate back